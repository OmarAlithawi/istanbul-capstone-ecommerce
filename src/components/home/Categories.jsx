import React from "react";
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";
import "./categories.scss";
import sportImg from "../../assets/img/dumbbell.jpg";
import glassesImg from "../../assets/img/glasses.jpg";
import setUpImg from "../../assets/img/setup.jpg";
import shoesImg from "../../assets/img/shoes.jpg";
import sterhoscopeImg from "../../assets/img/sterhoscope.jpg";
import NutritionImg from "../../assets/img/supplements.jpg";
import selfCareimg from "../../assets/img/selfCare.jpg";

const SLIDER_SETTINGS = {
  centerMode: true,
  centerPadding: "40px",
  autoplay: false,
  autoplaySpeed: 10000,
  speed: 1000,
  arrows: true,
  slidesToShow: 5,
  slidesToScroll: 1,
  dots: true,
  responsive: [
    {
      breakpoint: 768,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 1,
        arrows: false,
      },
    },
    {
      breakpoint: 500,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
        arrows: false,
      },
    },
  ],
};

const CATEGORIES_INFO = [
  {
    name: "Sports",
    img: sportImg,
  },
  {
    name: "Glasses",
    img: glassesImg,
  },
  {
    name: "Set Up",
    img: setUpImg,
  },
  {
    name: "Shoes",
    img: shoesImg,
  },
  {
    name: "Health",
    img: sterhoscopeImg,
  },
  {
    name: "Nutrition",
    img: NutritionImg,
  },
  {
    name: "Personal Care",
    img: selfCareimg,
  },
];

export default function Categories() {
  const Category = (props) => {
    return (
      <div className="card">
        <img alt="ctegory" src={`${props.category.img}`} />
        <p className="cardTitle">{props.category.name.toLocaleUpperCase()}</p>
      </div>
    );
  };

  return (
    <div className="categoriesContainer">
      <h1 className="categoryTitle">Categories</h1>

      <Slider {...SLIDER_SETTINGS}>
        {CATEGORIES_INFO.map((cat) => (
          <Category category={cat} key={cat.name} className="row" />
        ))}
      </Slider>
    </div>
  );
}
