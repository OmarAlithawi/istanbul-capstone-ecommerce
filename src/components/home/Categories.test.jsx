import React from 'react';
import Categories from './Categories';
import { render } from '@testing-library/react';

it('renders correctly when there are no items', async () => {
	const { findAllByAltText } = render(<Categories />);
	const items = await findAllByAltText('image');
	expect(items).toMatchSnapshot();
});
