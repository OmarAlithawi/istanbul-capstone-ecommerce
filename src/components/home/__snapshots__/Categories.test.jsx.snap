// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
Array [
  <img
    alt="image"
    src="glasses.jpg"
  />,
  <img
    alt="image"
    src="setup.jpg"
  />,
  <img
    alt="image"
    src="shoes.jpg"
  />,
  <img
    alt="image"
    src="sterhoscope.jpg"
  />,
  <img
    alt="image"
    src="supplements.jpg"
  />,
  <img
    alt="image"
    src="selfCare.jpg"
  />,
  <img
    alt="image"
    src="dumbbell.jpg"
  />,
  <img
    alt="image"
    src="glasses.jpg"
  />,
  <img
    alt="image"
    src="setup.jpg"
  />,
  <img
    alt="image"
    src="shoes.jpg"
  />,
  <img
    alt="image"
    src="sterhoscope.jpg"
  />,
  <img
    alt="image"
    src="supplements.jpg"
  />,
  <img
    alt="image"
    src="selfCare.jpg"
  />,
  <img
    alt="image"
    src="dumbbell.jpg"
  />,
  <img
    alt="image"
    src="glasses.jpg"
  />,
  <img
    alt="image"
    src="setup.jpg"
  />,
  <img
    alt="image"
    src="shoes.jpg"
  />,
  <img
    alt="image"
    src="sterhoscope.jpg"
  />,
  <img
    alt="image"
    src="supplements.jpg"
  />,
  <img
    alt="image"
    src="selfCare.jpg"
  />,
]
`;
